packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -197,11 +370,98 @@ export class Table extends Resource {
       actions: ['dynamodb:ListStreams'],
       resourceArns: ['*'],
     });
- }
+  }
+
+  /**
+   * Creates a Table construct that represents an external table via table name.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param tableName The table's name.
+   */
+  public static fromTableName(scope: Construct, id: string, tableName: string): ITable {
+    return Table.fromTableAttributes(scope, id, { tableName });
+  }
+
+  /**
+   * Creates a Table construct that represents an external table via table arn.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param tableArn The table's ARN.
+   */
+  public static fromTableArn(scope: Construct, id: string, tableArn: string): ITable {
+    return Table.fromTableAttributes(scope, id, { tableArn });
+  }
+
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs A `TableAttributes` object.
+   */
+  public static fromTableAttributes(scope: Construct, id: string, attrs: TableAttributes): ITable {
+
+    class Import extends TableBase {
+      public readonly tableName: string;
+      public readonly tableArn: string;
+
+      constructor(_scope: Construct, _id: string, props: TableAttributes) {
+        super(_scope, _id);
+        this.tableArn = props.tableArn!;
+        this.tableName = props.tableName!;
+      }
+
+      protected get hasIndex(): boolean {
+        return false;
+      }
+    }
+
+    const stack = Stack.of(scope);
+    let tableName = '';
+    let tableArn = '';
+    // if we have an explicit table name, use it.
+    if (attrs.tableName) {
+      tableName = attrs.tableName;
+      if (!attrs.tableArn) {
+        tableArn = stack.formatArn({
+          region: stack.region,
+          account: stack.account,
+          service: 'dynamodb',
+          resource: `table/${attrs.tableName}`
+        });
+      }
+    }
+
+    if (attrs.tableArn) {
+      tableArn = attrs.tableArn;
+      const arn = stack.parseArn(tableArn);
+      if (!arn.account || !arn.region) {
+        throw new Error(`Arn '${tableArn}' of DynamoDB table must be account and region sepcific resource.`);
+      }
+      if (!attrs.tableName) {
+        tableName = stack.parseArn(attrs.tableArn).resourceName!;
+      }
+    }
+
+    if (!attrs.tableName && !attrs.tableArn) {
+      throw new Error('Cannot determine table name. At least `tableArn` or `tableName` is needed');
+    }
+
+    if (!tableArn!.endsWith(tableName!)) {
+      throw new Error(`Given table name '${attrs.tableName}' conflicts with table arn '${attrs.tableArn}'.`);
```

Comment:
I appreciate these checks, but I wouldn't bother with them. Also, providing both `tableName` and `tableArn` should result in an error. Given that, I would re-write this method as:

```ts
public static fromTableAttributes(scope: Construct, id: string, attrs: TableAttributes): ITable {
  class Import extends TableBase {
    // ...
  } 

  let tableName: string;
  let tableArn: string;
  const stack = Stack.of(scope);
  if (!attrs.tableName) {
    if (!attrs.tableArn) { throw new Error('One of tableName or tableArn is required!'); }
    
    tableArn = attrs.tableArn;
    const maybeTableName = stack.formatArn(attrs.tableArn).resourceName;
    if (!maybeTableName) { throw new Error('ARN for DynamoDB table must be in the form: ...'); }
    tableName = maybeTableName;
  } else {
    if (attrs.tableArn) { throw new Error("Only one of tableArn or tableName can be provided"); }
    tableName = attrs.tableName;
    tableArn = stack.formatArn({
      service: 'dynamodb',
      resource: 'table',
      resourceName: attrs.tableName,
    });
  }

  return new Import(scope, id, { tableName, tableArn });
}
```